hardware/ASyncTCP.cpp

```diff
@@ -260,6 +310,18 @@ void ASyncTCP::write_end(const boost::system::error_code& error)
 				StartReconnect();
 			}
 		}
+		else {
+			boost::unique_lock<boost::mutex> lock(writeMutex);
+			if (writeQ.size() > 0) {
```

Comment:
Should this be a 'while > 0' ? could it be that there are more items in writeQ ?
Please use if (!writeQ.empty())

Reply:
writeQ,empty() used now. The other comment was a negative.